routes/baseDBMethods.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+
+function handleErrors(ctx, error) {
+  console.error(error);
+  if (error.name === 'ValidationError') {
+    ctx.badRequest('Invalid document');
+  } else if (error.name === 'CastError') {
+    // CastError was thrown since the given ID is not a valid ObjectId
+    ctx.badRequest(`Invalid id ${ctx.params.id}`);
+  } else {
+    ctx.internalServerError();
+  }
+}
+
+function handleDocResponse(ctx, doc) {
+  if (doc === null) {
+    ctx.notFound(`Document with id ${ctx.params.id} not found`);
+  } else {
+    ctx.ok(doc);
+  }
+}
+
+async function list(ctx, Model) {
+  // TODO pagination
+  try {
+    const docs = await Model.find({});
+    ctx.ok(docs);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function getOne(ctx, Model) {
+  try {
+    const doc = await Model.findById(ctx.params.id);
+    handleDocResponse(ctx, doc);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function create(ctx, Model) {
+  const requestedDoc = new Model(ctx.request.body);
+
+  try {
+    const savedDoc = await requestedDoc.save({
+      validateBeforeSave: true,
+    });
+    ctx.ok(savedDoc);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function update(ctx, Model) {
+  try {
+    const updatedDoc = await Model.findByIdAndUpdate(ctx.params.id,
+      { $set: ctx.request.body },
+      { new: true });
+    handleDocResponse(ctx, updatedDoc);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function remove(ctx, Model) {
+  try {
+    const removedDoc = await Model.findByIdAndRemove(ctx.params.id);
```

Comment:
Why did you changed all my functions?
I don't think it will work, in every case of 'findByIdAndRemove'/'findOneAndRemove' it removed randomly until all the collection was deleted unless I gave him the format: '_id': ctx.params.id like my original function

Reply:
It does work (I tested it).
Why would it remove randomly? It specifically removes the one document with the given id, just as before.
The usage of `findOneAndRemove`/`findByIdAndRemove` is different, but the former calls the latter anyway (as declared in mongoose documentation).